### PR TITLE
Typescript refactor

### PR DESCRIPTION
**Pull request by dochiasebastian**

### CODE REVIEW

File: 1. Basic Web Development/script.js [hunk 1]
```diff
@@ -1,90 +1,70 @@
-const API_URL = "http://localhost:3000";
-const HEADERS = {
-    "Content-Type": "application/json",
+var API_URL = "http://localhost:3000";
```
Comment (anbalase) [inline 1. Basic Web Development/script.js]: why var?

File: 1. Basic Web Development/script.js [hunk 2]
```diff
@@ -1,90 +1,70 @@
...
-            console.log(event.target.value);
+function handleRadios(radios, boxes) {
+    var _this = this;
+    radios.forEach(function (radio) {
+        radio.addEventListener('change', function (event) {
             switch (event.target.value) {
                 case "All":
-                    changeBoxesState(boxes.pmsBoxes, true);
-                    changeBoxesState(boxes.allBoxes, true);
+                    _this.changeBoxesState(boxes.pmsBoxes, true);
```
Comment (anbalase) [inline 1. Basic Web Development/script.js]: why?

File: 1. Basic Web Development/script.js [hunk 3]
```diff
@@ -1,90 +1,70 @@
...
-
     handlePopUp();
 });
-
-handleRadios = (radios, boxes) => {
-    radios.forEach(radio => {
-        radio.addEventListener('change', event => {
-            console.log(event.target.value);
+function handleRadios(radios, boxes) {
+    var _this = this;
```
Comment (anbalase) [inline 1. Basic Web Development/script.js]: This is a code-smell 😄

File: 1. Basic Web Development/server/dist/server.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+"use strict";
```
Comment (anbalase) [inline 1. Basic Web Development/server/dist/server.js]: let's exclude compiled files in `.gitignore`

File: 1. Basic Web Development/server/dist/server.js.map [hunk 5]
```diff
@@ -0,0 +1 @@
+{"version":3,"file":"server.js","sourceRoot":"","sources":["../src/server.ts"],"names":[],"mappings":";;AAAA,mCAAoC;AAEpC,6BAA8B;AAE9B,MAAM,GAAG,GAAG,OAAO,EAAE,CAAC;AAEtB,GAAG,CAAC,GAAG,CAAC,IAAI,EAAE,CAAC,CAAC;AAEhB,GAAG,CAAC,GAAG,CAAC,OAAO,CAAC,IAAI,EAAE,CAAC,CAAC;AAExB,GAAG,CAAC,IAAI,CAAC,GAAG,EAAE,CAAC,GAAG,EAAE,GAAG,EAAE,EAAE;IACvB,OAAO,CAAC,GAAG,CAAC,GAAG,CAAC,IAAI,CAAC,CAAC;IAEtB,GAAG,CAAC,MAAM,CAAC,6BAA6B,EAAE,GAAG,CAAC,CAAC;IAC/C,GAAG,CAAC,MAAM,CAAC,8BAA8B,EAAE,2BAA2B,CAAC,CAAC;IACxE,GAAG,CAAC,MAAM,CAAC,8BAA8B,EAAE,+DAA+D,CAAC,CAAC;IAE5G,GAAG,CAAC,MAAM,CAAC,GAAG,CAAC,CAAC,IAAI,CAAC;QACjB,OAAO,EAAE,IAAI;QACb,IAAI,EAAE,GAAG,CAAC,IAAI;KACjB,CAAC,CAAC;AACP,CAAC,CAAC,CAAC;AAEH,GAAG,CAAC,MAAM,CAAC,IAAI,EAAE,GAAG,EAAE;IAClB,OAAO,CAAC,GAAG,CAAC,6BAA6B,CAAC,CAAA;AAC9C,CAAC,CAAC,CAAC"}
```
Comment (anbalase) [inline 1. Basic Web Development/server/dist/server.js.map]: add to `.gitignore`

File: 1. Basic Web Development/script.js (hunk 1 above)
Comment (dochiasebastian) [reply 1. Basic Web Development/script.js]: This is how the compiler translated it from TS.

File: 1. Basic Web Development/script.js (hunk 2 above)
Comment (dochiasebastian) [reply 1. Basic Web Development/script.js]: Same here

File: 1. Basic Web Development/script.js (hunk 3 above)
Comment (dochiasebastian) [reply 1. Basic Web Development/script.js]: Same here

File: 1. Basic Web Development/server/dist/server.js (hunk 4 above)
Comment (dochiasebastian) [reply 1. Basic Web Development/server/dist/server.js]: I'll do that right now with both the frontend and backend

File: 1. Basic Web Development/server/dist/server.js.map (hunk 5 above)
Comment (dochiasebastian) [reply 1. Basic Web Development/server/dist/server.js.map]: I tried with all my might and failed miserably.

**Review by anbalase — APPROVED** [review]

**Pull request merged**